b2luigi/batch/processes/htcondor.py

```diff
@@ -0,0 +1,225 @@
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._batch_job_id = None
+
+    def get_job_status(self):
+
+        assert self._batch_job_id
+
+        try:
+            job_status = _batch_job_status_cache[self._batch_job_id]
+        except KeyError:
+            return JobStatus.aborted
+        if job_status == HTCondorJobStatus.completed:
+            return JobStatus.successful
+        elif job_status == HTCondorJobStatus.idle:
+            return JobStatus.running
+        elif job_status == HTCondorJobStatus.running:
+            return JobStatus.running
+        elif job_status == HTCondorJobStatus.removed:
+            return JobStatus.aborted
+        elif job_status == HTCondorJobStatus.held:
+            return JobStatus.aborted
+        elif job_status == HTCondorJobStatus.submission_err:
+            return JobStatus.aborted
+        else:
+            raise ValueError(f"Unknown HTCondor Job status: {job_status}")
+
+    def start_job(self):
+        submit_file_path = self._create_htcondor_submit_file()
+        exec_cmd, curr_env = self._create_condor_executable_cmd()
+
+        submit_file_dir, submit_file = os.path.split(submit_file_path)
+        cmd = ["condor_submit", submit_file]
+
+        # apparently you have to in the same directory as the submit file
+        # to be able to submit jobs with htcondor
+        cur_dir = os.getcwd()
+
+        # have to copy settings file to job working directory since b2luigi has to
+        # take the result path from it
+        shutil.copyfile("settings.json", os.path.join(submit_file_dir, "settings.json"))
+        os.chdir(submit_file_dir)
+        output = subprocess.check_output(cmd, env=curr_env)
+        os.chdir(cur_dir)
+        output = output.decode()
+        match = re.search(r"[0-9]+\.", output)
+        if not match:
+            raise RuntimeError("Batch submission failed with output " + output)
+
+        self._batch_job_id = int(match.group(0)[:-1])
+
+    def kill_job(self):
+        if not self._batch_job_id:
+            return
+        subprocess.check_call(["condor_rm", str(self._batch_job_id)], stdout=subprocess.DEVNULL)
+
+    def _create_htcondor_submit_file(self):
+        output_path = get_output_dirs(self.task, create_folder=True)
+        submit_file_path = os.path.join(output_path, "job.submit")
+        executable_wrapper_path = self._create_executable_wrapper()
+
+        log_file_dir = get_log_file_dir(self.task)
+        stderr_log_file = log_file_dir + "job.err"
+        stdout_log_file = log_file_dir + "job.out"
+        job_log_file = log_file_dir + "job.log"
+
+        general_settings = get_setting("htcondor_settings", dict())
+
+        try:
+            job_settings = self.task.htcondor_settings
+        except AttributeError:
+            job_settings = {}
+
+        with open(submit_file_path, "w") as submit_file:
+            submit_file.write(f"executable = {executable_wrapper_path}\n")
```

Comment:
Maybe worth having a multiline string here?
```python
submit_file.write(f"""
executable = ...
...
    """)

Reply:
Added the multiline string form. Is there any other advantage of doing it this way than that it"s easier to read?

Reply:
No, really just readability. Thanks!